src/transport_performance/osm/validate_osm.py

```diff
@@ -0,0 +1,739 @@
+            "relation_ids": self.__relations_ids,
+            "area_ids": self.__area_ids,
+        }
+        self.id_dict = id_dict
+        return id_dict
+
+
+class FindTags:
+    """Applies tag collation to OSM file.
+
+    Parameters
+    ----------
+    osm_pth : Union[Path, str]
+        Path to osm file.
+    tag_collator : _TagHandler, optional
+        FindTags applies the logic from tag_collator to a pbf file on init,
+        storing the collated tags in a `found_tags` attribute. Defaults to
+        _TagHandler.
+
+    Raises
+    ------
+    TypeError :
+        `osm_pth` is not of type pathlib.Path or str.
+    FileNotFoundError :
+        `osm_pth` file not found on disk.
+    ValueError :
+        `osm_pth` does not have a .pbf extension.
+
+    Attributes
+    ----------
+    found_tags : dict
+        Found tags for specified feature IDs.
+    __node_tags : dict
+        Tags found for OSM node features.
+    __way_tags : dict
+        Tags found for OSM way features.
+    __relation_tags : dict
+        Tags found for OSM relation features.
+    __area_tags : dict
+        Tags found for OSM area features.
+
+    Methods
+    -------
+    check_tags_for_ids()
+        Filter tags to the given list of IDs. Updates `found_tags` attribute.
+
+    """
+
+    def __init__(
+        self, osm_pth, tag_collator: _TagHandler = _TagHandler
+    ) -> None:
+        _is_expected_filetype(
+            osm_pth, "osm_pth", check_existing=True, exp_ext=".pbf"
+        )
+        tags = _TagHandler()
+        tags.apply_file(osm_pth)
+        self.__node_tags = tags.node_tags
+        self.__way_tags = tags.way_tags
+        self.__relation_tags = tags.relation_tags
+        self.__area_tags = tags.area_tags
+        self.found_tags = dict()
+
+    def check_tags_for_ids(self, ids: list, feature_type: str) -> dict:
+        """Return tags for provided list of feature IDs.
+
+        Parameters
+        ----------
+        ids : list
+            A list of OSM feature IDs to check. IDs must be integer.
+        feature_type : str
+            The type of feature to which the IDS belong. Valid options are
+            "node", "way", "relation", or "area".
+
+        Returns
+        -------
+        found_tags : dict
+            Dictionary of tag IDs and their tag data in ID : {name:value}
+            format.
+
+        """
+        self.found_tags = _filter_target_dict_with_list(
+            targets={
+                "node": self.__node_tags,
+                "way": self.__way_tags,
+                "relation": self.__relation_tags,
+                "area": self.__area_tags,
+            },
+            _list=ids,
+            search_key=feature_type,
+        )
+        return self.found_tags
+
+
+class FindLocations:
+    """Applies location collation to OSM file.
+
+    Parameters
+    ----------
+    osm_pth : Union[Path, str]
+        Path to osm file.
+    loc_collator : _LocHandler=_LocHandler
```

Comment:
Type in the docstring here. `=_LocHandler` can be removed and replaced with `loc_collator : _LocHandler, optional`

Reply:
Done.